src/java/org/apache/cassandra/net/AbstractMessageHandler.java

```diff
@@ -311,7 +311,10 @@ private void onReserveCapacityRegained(Limit endpointReserve, Limit globalReserv
                 decoder.reactivate();
 
                 if (decoder.isActive())
-                    ClientMetrics.instance.unpauseConnection();
```

Comment:
Alternative approach would be to check `initialized` in all metrics access methods, eg. `unpauseConnection`, but that may hide some bugs: `ClientMetrics` was not initialized when it should be initialized.

Reply:
Yeah I also thought about that and had the same feeling about potential hidden bugs.